cpp/src/reductions/simple_segmented.cuh

```diff
@@ -0,0 +1,271 @@
+    }
+  }();
+
+  // Compute the output null mask
+  auto const bitmask                 = col.null_mask();
+  auto const first_bit_indices_begin = offsets.begin<size_type>();
+  auto const first_bit_indices_end   = offsets.end<size_type>() - 1;
+  auto const last_bit_indices_begin  = offsets.begin<size_type>() + 1;
+  auto const [output_null_mask, output_null_count] =
+    cudf::detail::segmented_null_mask_reduction(bitmask,
+                                                first_bit_indices_begin,
+                                                first_bit_indices_end,
+                                                last_bit_indices_begin,
+                                                null_handling,
+                                                stream,
+                                                mr);
+  result->set_null_mask(output_null_mask, output_null_count, stream);
+
+  return result;
+}
+
+/**
+ * @brief Call reduce and return a column of type bool.
+ *
+ * This is used by operations `any()` and `all()`.
+ *
+ * @tparam Op The reduce operation to execute on the column.
+ */
+template <typename Op>
+struct bool_result_column_dispatcher {
+  template <typename ElementType,
+            std::enable_if_t<std::is_arithmetic<ElementType>::value>* = nullptr>
+  std::unique_ptr<column> operator()(column_view const& col,
+                                     column_view const& offsets,
+                                     null_policy null_handling,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr)
+  {
+    return simple_segmented_reduction<ElementType, bool, Op>(
+      col, offsets, null_handling, stream, mr);
+  }
+
+  template <typename ElementType,
+            std::enable_if_t<not std::is_arithmetic<ElementType>::value>* = nullptr>
+  std::unique_ptr<column> operator()(column_view const&,
+                                     column_view const&,
+                                     null_policy,
+                                     rmm::cuda_stream_view,
+                                     rmm::mr::device_memory_resource*)
+  {
+    CUDF_FAIL("Reduction operator not supported for this type");
+  }
+};
+
+/**
+ * @brief Call reduce and return a column of type matching the input column.
+ *
+ * This is used by operations `min()` and `max()`.
+ *
+ * @tparam Op The reduce operation to execute on the column.
+ */
+template <typename Op>
+struct same_column_type_dispatcher {
+ private:
+  template <typename ElementType>
+  static constexpr bool is_supported()
+  {
+    return !(cudf::is_dictionary<ElementType>() || std::is_same_v<ElementType, cudf::list_view> ||
+             std::is_same_v<ElementType, cudf::struct_view>);
+  }
+
+ public:
+  template <typename ElementType,
+            std::enable_if_t<is_supported<ElementType>() &&
+                             not cudf::is_fixed_point<ElementType>()>* = nullptr>
```

Comment:
It looks like the `is_supported` function should also exclude fixed point values, rather than excluding them separately in the SFINAE templates.